docs/dpg-archboard.md

```diff
@@ -0,0 +1,108 @@
+# Python DPG Client
+
+## Intro
+
+## Breaking Changes
+
+* If the service doesn't break, the client shouldn't break.
+* The main steps we've taken to mitigate this is using
+    * For now, we're generating without models, and are going to be dealing with raw JSON. The feedback is actually positive for Python, Python users are very
+    comfortable dealing with raw JSON, and in many cases prefer it. We are looking into our DPG models, which will be non-breaking, and provide additional
+    documentation sticking points for people to look at
+    * Make header and query parameters keyword only.
+
+## Developer Experience
+
+### Creating a client
+
+```python
+from azure.purview.catalog import PurviewCatalogClient
+from azure.identity import DefaultAzureCredential
+
+client = PurviewCatalogClient(endpoint="http://myendpoint.com", credential=DefaultAzureCredential())
+```
+
+### Simple GET request
+
+```python
+response = client.my_get_call()
+```
+
+### Creating a POST request
+
+For model / dictionary / list input JSON types, we've added an overload for users to use. We don't want to force users to read in large models into memory
+if the model is big, so we give users the option to pass in a file descriptor of serialized json as well.
+
+```python
+
+client.put_object({"hello": "world"})
+
+with open("myLargeJsonFile", "rb") as fd:
+    client.put_object(fd)
+```
+
+### Working with the response
+
+#### JSON responses
+
+We return JSON responses back to users
+
+```python
+response = client.get_object()
+assert response["hello"] == "world!"
+```
+
+#### Stream responses
+
+```python
+response = client.get_streamed_response()
+for data in response.iter_bytes():
+    print(data)
```

Comment:
I haven't ever meet this kind of scenario. Is there any practical example?

Reply:
I honestly don't know of one in our real SDKs, the scenario would be something like getting a streamed file from a service. We do have tests for this, it's in vanilla/BodyFile, here's our sync acceptance test https://github.com/Azure/autorest.python/blob/autorestv3/test/vanilla/version-tolerant/AcceptanceTests/test_file.py